src/librustc_typeck/diagnostics.rs

```diff
@@ -12,6 +12,28 @@
 
 register_long_diagnostics! {
 
+E0046: r##"
+When trying to make some type implement a trait `Foo`, you must, at minimum,
+provide implementations for all of `Foo`'s required methods (meaning the
+methods that do not have default implementations), as well as any required
+trait items like associated types or constants.
+"##,
+
+E0054: r##"
+It is not allowed to cast to a bool. If you are trying to cast a numeric type
```

Comment:
From my (admittedly shallow) reading of the code in librustc_typeck/check/cast.rs, it seemed like any casting to bool is disallowed. For example, you can try to cast a bare function to a bool and still E0054.

```rust
fn main() {
    fn foo() -> u8 { 5 }
    let x = foo as bool; // error: cannot cast as `bool`, compare with zero instead [E0054]
}
```